### ISSUE TITLE
Refactor data processing pipeline

### ISSUE THREAD

**Issue by ha0ye** [weecology/portalr#209]
Some of the `group_by` + `summarize` steps are very slow on certain OS implementations. (I'm not sure exactly why, but maybe due to how dplyr is optimized under the hood?

This seems to be the cause of some vignettes taking a long time to build, as well as tests timing out ~ #208.

At least one instance is `make_level_data` doesn't need to group_by and summarize if `level == "plot"`, since the data is already aggregated at the plot & species level -- we need only mutate accordingly.

**Comment by ha0ye** [comment]
For large number of groups (~100,000) and some summarize functions, processing is very slow on my windows machine. I'm not sure what might be the issue (depends on what operations are handled well by the underlying c++ code in dplyr), but worth pointing this out.

**Comment by ha0ye** [comment]
Ok, it seems maybe unrelated to summarize, but math operations while grouped (that aren't being appropriately vectorized)?

```r
library(dplyr)
n <- 50000
df <- data.frame(a = 1:n, x = rnorm(n))

tictoc::tic()
df %>% group_by(a) %>% mutate(x = x + 10)
tictoc::toc()
```

takes 138.65 seconds on my windows desktop and 0.441 seconds on my mac laptop!! 🙀

**Comment by ha0ye** [comment]
Related to dplyr 0.8.2. slowdown, that is now fixed in 0.8.3. (refactoring already done)
